pallets/creditcoin/src/migrations/v8.rs

```diff
@@ -0,0 +1,109 @@
+use super::Migrate;
+use super::{vec, Vec};
+
+use crate::types::CollectedCoinsStruct;
+
+use crate::{AddressId, Config};
+use crate::{CollectedCoinsId, ExternalTxId};
+use frame_support::weights::Weight;
+use frame_support::{pallet_prelude::*, traits::Get};
+use parity_scale_codec::{Decode, Encode};
+
+#[derive(Clone, Encode, Decode)]
+pub struct OldCollectedCoinsStruct<Hash, Balance> {
+	pub to: AddressId<Hash>,
+	pub amount: Balance,
+	pub tx_id: ExternalTxId,
+}
+
+pub(super) struct Migration<Runtime>(PhantomData<Runtime>);
+
+impl<Runtime: Config> Migration<Runtime> {
+	pub(super) fn new() -> Self {
+		Self(PhantomData)
+	}
+}
+
+impl<T: Config> Migrate for Migration<T> {
+	fn pre_upgrade(&self) -> Vec<u8> {
+		vec![]
+	}
+
+	fn post_upgrade(&self, _blob: Vec<u8>) {
+		assert_eq!(
+			StorageVersion::get::<crate::Pallet<T>>(),
+			8,
+			"expected storage version to be 8 after migrations complete"
+		);
+	}
+	fn migrate(&self) -> Weight {
+		let mut weight: Weight = Weight::zero();
+
+		let weight_each = T::DbWeight::get().reads_writes(1, 1);
+
+		crate::CollectedCoins::<T>::translate::<OldCollectedCoinsStruct<T::Hash, T::Balance>, _>(
```

Comment:
This migration for `CollectedCoins` looks right, but since you modified `UnverifiedCollectedCoins` you need to migrate pending tasks as well. So after you've copied the types from before this PR into v6, you can just `use super::v6::UnverifiedCollectedCoins as OldUnverifiedCollectedCoins` and likewise for `Task`. 

Similar to my comment in `v6.rs`, this is awkward because the `PendingTasks` storage item is in the task scheduler pallet and sort of half abstracted away by the `TaskScheduler` trait.

(As an aside, this all has made me realize that it probably isn't a good idea to have pallets that have storage items that hold types that live outside that pallet. For instance, in this case, the creditcoin pallet defines the concrete types held by the `PendingTasks` storage item, which is currently in the task scheduler pallet. I feel like it makes more sense for that storage to live in the creditcoin pallet. You can't really abstract away the storage item fully, like the `TaskScheduler` trait tries to do, because migrations need to access them directly which breaks the abstraction).

I think the easiest solution is to once again just access it directly by setting up a storage alias, like in `v6` or the other migrations, and then `translate` the `PendingTasks` to the new versions.

Reply:
Resolving since there's an open ticket